R/tree-utils.R

```diff
@@ -6,9 +6,12 @@ generate_top_level_map <- function(pc) {
   tl_parent <- pc$parent
   tl_parent[pc$parent <= 0L] <- tl_ids
   i_not_assigned <- which(!tl_parent %in% tl_ids)
-  while (length(i_not_assigned)) {
+  while ((prev_length <- length(i_not_assigned)) > 0L) {
     tl_parent[i_not_assigned] <- pc$parent[match(tl_parent[i_not_assigned], pc$id)]
     i_not_assigned <- which(!tl_parent %in% tl_ids)
+    if (length(i_not_assigned) >= prev_length) {
+      stop("Logical error: unassigned set did not shrink. Check file syntax and please report as a bug.")
```

Comment:
Add GitHub issue link? As-is the error message might not be clearly linked to lintr in production.

Reply:
Just added 'lintr' -- should be slightly more stable than referencing the GH link (GH site structure changes, or "turns evil" & we change hosts, etc)